### ISSUE TITLE
Project issues containing images overflow(-x) theirs columns

### ISSUE THREAD

**Issue by Simandara** [go-gitea/gitea#31667]
### Description

It is possible to insert images in issues as attachement. When such issues are displayed in Projects, the card often display some images on top of the issue. This is practical as it allow to peek issues by images and illustrations rather than reading titles.

Sometimes, images in cards are resized to fit in, and sometime they are full-sized.

# Issue

However, sometime they are wider than the card, and overflow on the column containing the card (on the x axis).
- It display an horizontal scrollbar on the bottom of the column
- It move all cards of a column at once when scrolling horizontally
- It feel weird
<img width="347" alt="Capture d’écran 2024-07-21 à 02 01 27" src="https://github.com/user-attachments/assets/c8df1097-d343-464b-91b8-71a27bbad03f">

Also, the images are overflowing the card while dragging it:
<img width="864" alt="Capture d’écran 2024-07-21 à 02 00 52" src="https://github.com/user-attachments/assets/7863bef7-f832-4ca2-ad64-d49c26ed8e4e">

# Suggestion

Make the images scrolls in the card instead:
![Enregistrementdelecran2024-07-21a03 03 56-ezgif com-video-to-gif-converter](https://github.com/user-attachments/assets/cc802d12-a5cc-4dd2-ab3f-8ade0964941a)
- Scrollbars only in card having overflowing images
- Only scroll images in cards, not all cards of a column
- Scroll snap to the closest image
- Overlapping scrollbars on some OS and browsers (Windows 10<, Some Linux distros)
- Card properly framed while dragging

<img width="356" alt="Capture d’écran 2024-07-21 à 02 52 14" src="https://github.com/user-attachments/assets/6003d1b0-9142-4490-a3ce-93452497b7b3">

To create this effect I changed the following:
```CSS
.issue-card {
  /* [...] */
  align-items:stretch;
}

.card-attachment-images {
  /* [...] */
  overflow:scroll;
  cursor: default;
  scroll-snap-type: x mandatory;
}

.card-attachment-images img {
  /* [...] */
  text-align: left;
  scroll-snap-align: center;
}
```

What do you think ?

### Screenshots

<img width="347" alt="Capture d’écran 2024-07-21 à 02 01 27" src="https://github.com/user-attachments/assets/c8df1097-d343-464b-91b8-71a27bbad03f">

### Gitea Version

1.22.0

### Can you reproduce the bug on the Gitea demo site?

No

### Browser Version

Firefox 128

**Comment by yp05327** [comment]
Can you create a PR to fix it?
I have done something last year in #26561, but it is.... stopped.

**Comment by SimonPistache** [comment]
> Can you create a PR to fix it? I have done something last year in #26561, but it is.... stopped.

I'll have a look this week, toying with devtools is a thing, changing the source code is another :)

**Comment by yp05327** [comment]
Never mind. If you want to, then just try it. You can ask for help at any time.

**Comment by SimonPistache** [comment]
Thanks for the support, i'll give it a shot then. 🙂

What about the idea? Having scrollbar on images on card is ok to all?

**Comment by yp05327** [comment]
I think it is not bad, so I suggested you to create a PR.
But I suggest adding some pictures like this: ▊
This may broke the vertical view I think.

**Comment by lunny** [comment]
Resolved by #31683
